posix/get-buildtool-lang

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+
+# Define language and their corresponding extensions using comma as delimiter
+langs="Java:java,Python:py,JavaScript:js,TypeScript:ts,C:c,C++:cpp,C++:cxx,C++:cc,CSharp:c#,CSharp:cs,PHP:php,Golang:go,Rust:rs,Kotlin:kt,Kotlin:kts,Lua:lua,Dart:dart,Ruby:rb,Swift:swift,R:r,VBScript:vb,Groovy:groovy,Scala:scala,Perl:pl,Godot:gd,Objective-C:m,Elixir:exs,Haskell:hs,Pascal:pas,Lisp:lisp,Lisp:cl,Lisp:clj,Lisp:cljs,Julia:jl,Zig:zig,Fortran:f,Solidity:sol,Ada:adb,Erlang:erl,Erlang:hrl,F#:fs,Apex:cls,Prolog:pro,OCaml:ml,COBOL:cbl,COBOL:cob,Crystal:cr,Nim:nim,Assembly:asm,VBA:vba,Shell:sh,Shell:bash,PowerShell:ps1,SQL:sql,HTML:html,HTML:htm,CSS:css"
+
+# Function to detect languages based on file extensions in the given directory and its subdirectories
+detect_languages() {
+    repo_path="$1"
+    detected_langs=""
+
+    # Get all file extensions in the repository
+    files=$(find "$repo_path" -type f -not -path '*/\.*' 2>/dev/null | sed -E 's/.*\.(.*)$/\1/' | sort | uniq)
```

Comment:
This gives filenames that don't have any extensions which can be a lot and take time to compute. Also there's no maxdepth so this can take longer for large repos with too much nested directories. We should use some maxlimit such as 7 (to account for languages like Java which has files in package structures).

Check time on harness-core:
`time find "$repo_path" -type f -maxdepth 7 -name '*.*' -not -path '*/\.*' | grep -E '\.[a-zA-Z0-9]+$' | sed -n 's/.*\.//p' | sort | uniq 2> /dev/null`

Reply:
harness core took 1.8sec
real    0m1.864s
user    0m0.067s
sys     0m0.616s

Reply:
It's always good to limit the maxdepth. Also only only files that have extensions (`*.*`).
`time find "$repo_path" -type f -maxdepth 7 -name '*.*' | grep -E '\.[a-zA-Z0-9]+$' | sed -n 's/.*\.//p' | sort | uniq 2> /dev/null` took less than a second for example.

Reply:
with depth 7 it took around 0.5 seconds , made the changes